application/src/main/java/org/togetherjava/tjbot/commands/tags/TagCommand.java

```diff
@@ -62,4 +69,20 @@ public void onSlashCommand(SlashCommandInteractionEvent event) {
         }
         message.queue();
     }
+
+    @Override
+    public void onAutoComplete(CommandAutoCompleteInteractionEvent event) {
+        AutoCompleteQuery focusedOption = event.getFocusedOption();
+
+        if (focusedOption.getName().equals(ID_OPTION)) {
```

Comment:
1. use early-return to reduce the nesting
2. use fail-fast and not defensive programming here. u dont expect other options to end up here, so if it does, it should throw

Reply:
not resolved. u didnt do this at all

Reply:
which error to throw?

Reply:
`IllegalArgumentException` since u got an option that u didnt expect.